[LiSimeng] ip

simenggg opened the pull request:
### **Lemon**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Lemon frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- fast to use

All you need to do is,
1. Download from [here](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html)
2. Double-click it
3. Add your tasks
4. Let it manage the tasks for you! 
And it's **free**! 👍 🎉 

Features:

- [x] Managing tasks
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
~~~
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
~~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

simenggg committed:
rename, greet and exit

simenggg committed:
echo and bye

simenggg committed:
add, list

simenggg committed:
mark, unmark

simenggg committed:
add todo, deadline and event

simenggg committed:
handle errors:incorrect description of todo, deadline and event and wrong messages

simenggg committed:
delete a task

simenggg committed:
Check if a tasklist file is present, create one if not

Try to create an addtask method to avoid repetition of code for todo, deadline and event

gajinkee commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+
+            } else if(userInputSplitted[0].equals("unmark")) {
+                int taskIndex = Integer.parseInt(userInputSplitted[1]);
+                System.out.println("OK, I've marked this task as not done yet");
```
Maybe consider using switch case for readability as there are many else branches?

gajinkee commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,30 @@
+public class Task {
+    protected String description;
+    protected boolean isdone;
```
Should this be in camel case instead? (ie isDone)

gajinkee replied:
Maybe consider setting these to private as they should not be accessed outside the Task objects?

gajinkee commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task{
+    protected String start;
```
Should this be set to private as they should not be accessed outside of this object?

gajinkee commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task{
+    protected String by;
```
Maybe use a more descriptive and less misleading name as "by" could also mean who the task was created by?

gajinkee reviewed (commented):
LGTM. Just some nits to pick. I hope you are catching up well for the IP as the PR looks slightly behind, perhaps you forgot to do a `git push`?

RickyHFR commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task{
```
```suggestion
public class Deadline extends Task {
```
Maybe it is better to leave a space after class name for better clarity?

RickyHFR commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task{
```
```suggestion
public class Event extends Task {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+        Task[] tasks = new Task[100];
+        int index = 0;
+
+        while(true) {
```
```suggestion
        while (true) {
```
You can consider leaving a space after each if / while / for etc :)

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+            String userInput = scanner.nextLine();
+            String[] userInputSplitted = userInput.split(" ");
+
+            if(userInput.equals("bye")) {
```
```suggestion
            if (userInput.equals("bye")) {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+                System.out.println("——————————————————————————————————————————————");
+                break;
+
+            } else if(userInput.equals("list")) {
```
```suggestion
            } else if (userInput.equals("list")) {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+
+            } else if(userInput.equals("list")) {
+                for (int i = 1; i < index + 1; i++) {
+                    if(tasks[i - 1] != null) {
```
```suggestion
                    if (tasks[i - 1] != null) {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+                }
+
+
+            } else if(userInputSplitted[0].equals("mark")){
```
```suggestion
            } else if (userInputSplitted[0].equals("mark")){
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+                System.out.println(tasks[taskIndex].toString());
+
+
+            } else if(userInputSplitted[0].equals("unmark")) {
```
```suggestion
            } else if (userInputSplitted[0].equals("unmark")) {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+                System.out.println(tasks[taskIndex].toString());
+
+
+            } else if(userInputSplitted[0].equals("delete")) {
```
```suggestion
            } else if (userInputSplitted[0].equals("delete")) {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+                System.out.println(tasks[taskIndex - 1].toString());
+                tasks[taskIndex - 1] = null;
+                //problem: there will be empty holes in the array, need to move the tasks to fill in the hole
+                for(int i = taskIndex - 1; i < index + 1; i++) {
```
```suggestion
                for (int i = taskIndex - 1; i < index + 1; i++) {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+                System.out.println("——————————————————————————————————————————————");
+                int spaceindex = userInput.indexOf(" ");   //find the index of the first space
+
+                if(userInputSplitted[0].equals("todo")) {
```
```suggestion
                if (userInputSplitted[0].equals("todo")) {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+
+                if(userInputSplitted[0].equals("todo")) {
+                    String taskDescription = userInput.substring(spaceindex + 1);
+                    if(userInputSplitted.length < 2) {
```
```suggestion
                    if (userInputSplitted.length < 2) {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+                        tasks[index++] = newTask;
+                        System.out.println("Got it. I've added this task:");
+                        System.out.println(" " + newTask.toString());
+                        if(index == 1) {
```
```suggestion
                        if (index == 1) {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+                    String task = userInput.substring(spaceindex + 1);
+                    String[] taskSplitted = task.split("/");
+                    String taskDescription = taskSplitted[0];
+                    if(taskSplitted.length != 2 ) {
```
```suggestion
                    if (taskSplitted.length != 2 ) {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+
+                    }
+
+                } else if(userInputSplitted[0].equals("event")) {
```
```suggestion
                } else if (userInputSplitted[0].equals("event")) {
```

RickyHFR commented on src/main/java/Lemon.java:
```diff
@@ -0,0 +1,135 @@
+                    String[] taskSplitted = task.split("/");
+                    String taskDescription = taskSplitted[0];
+
+                    if(taskSplitted.length != 3) {
```
```suggestion
                    if (taskSplitted.length != 3) {
```

RickyHFR reviewed (commented):
LGTM. You just need to add some extra spaces XD. A side note is that you did not include any JavaDoc. You may consider including them to make your code fully align with the coding standard! Jiayous and keep it up!!

simenggg committed:
Update the changes in the tasklist to the file stored in hard disk
Change toString methods to Todo, Deadline and Event classes

simenggg committed:
Merge branch 'branch-Level-7'

simenggg committed:
Store the deadline date in deadline class as a LocalDate object

simenggg committed:
Merge branch 'branch-Level-8'

simenggg committed:
Add more classes: UI, Storage, Tasklist, Command and Parser
Change the representation of tasklist from array to arraylist

simenggg committed:
Fix some bugs

simenggg committed:
Merge branch 'A-MoreOOP'

simenggg committed:
Create package lemon and put all the files in lemon

simenggg committed:
Merge branch 'add-gradle-support'

simenggg committed:
Add Gradle

simenggg committed:
Add JUnit tests for Todo, Task and Deadline classes

simenggg committed:
Create Jar file

simenggg committed:
Add JavaDoc comments for most classes and methods

simenggg committed:
Make some changes to comply to coding standard

simenggg committed:
Find a task by searching for a keywork in the task description

simenggg committed:
resolve merge conflict

simenggg committed:
Merge branch 'branch-A-JavaDoc'

simenggg committed:
Add GUI

simenggg committed:
Add assertions

There are several assumptions made in the user input, such as the format of the input. Wrong format may cause the chatbot to be unable to interpret some commands

Adding assertions to record important assumptions in the code so that mistakes in the code can be easily recognized

simenggg committed:
Improve code quality

Shorten some methods by extracting some parts out to a new method. In this way, the code will be more easily understandable.

simenggg committed:
Merge pull request #3 from simenggg/branch-A-Assertions 

Add assertions

simenggg committed:
Merge branch 'master' of https://github.com/simenggg/ip

simenggg committed:
Resolve conflict when merging the CodeQuality branch with the master branch

simenggg committed:
Merge pull request #2 from simenggg/branch-A-CodeQuality

Improve code quality

simenggg committed:
Add extension C-Help

Provide a command by which the user can access a help page, including technical help and emotional support

Additionally, fix some mino bugs in delete, mark and unmark methods in tasklist class

simenggg committed:
Merge branch 'master' of https://github.com/simenggg/ip

simenggg committed:
Merge branch 'branch-BCD-Extension'

simenggg committed:
Write a user guide for the product

simenggg committed:
Add a representative screenshot of the product

simenggg committed:
Add screenshot in user guide

simenggg committed:
Make some final changes

simenggg committed:
Improve error handling for wrong command format

simenggg committed:
Handle invalid index for mark, unmark and delete

simenggg committed:
Add welcome message in GUI
